swap/src/bin/swap.rs

```diff
@@ -94,23 +94,24 @@ async fn main() -> Result<()> {
                 EventLoop::new(swap_id, swarm, alice_peer_id, bitcoin_wallet.clone())?;
             let event_loop = tokio::spawn(event_loop.run());
 
-            let send_bitcoin = determine_btc_to_swap(
-                event_loop_handle.request_quote(),
-                bitcoin_wallet.balance(),
-                bitcoin_wallet.new_address(),
-                async {
-                    while bitcoin_wallet.balance().await? == Amount::ZERO {
-                        bitcoin_wallet.sync().await?;
+            debug!("Requesting quote");
+            let bid_quote = event_loop_handle.request_quote().await?;
+            info!("Received quote: 1 XMR ~ {}", bid_quote.price);
```

Comment:
Why not leave that inside?

Reply:
I tried to depict more clearly what actually has to be value and what a future / function in the API of `determin_btc_to_swap`. 
We only call `determine_btc_to_swap` once so I did not see any reason not to move this outside. 
Don't have strong feelings about it though, can move it inside again :)

Reply:
Fine either way, your position also makes sense.